hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/connector/StreamEventJournalP.java

```diff
@@ -0,0 +1,422 @@
+                instance -> (EventJournalReader<EventJournalMapEvent>) instance.getMap(mapName),
+                predicate, projection, newestSequence);
+    }
+
+    public static <T> ProcessorMetaSupplier streamMap(String mapName,
+                                                      ClientConfig clientConfig,
+                                                      DistributedPredicate<EventJournalMapEvent> predicate,
+                                                      DistributedFunction<EventJournalMapEvent, T> projection,
+                                                      boolean newestSequence) {
+        return new RemoteClusterMetaSupplier<>(clientConfig,
+                instance -> (EventJournalReader<EventJournalMapEvent>) instance.getMap(mapName),
+                predicate, projection, newestSequence);
+    }
+
+    public static <T> ProcessorMetaSupplier streamCache(String cacheName,
+                                                        DistributedPredicate<EventJournalCacheEvent> predicate,
+                                                        DistributedFunction<EventJournalCacheEvent, T> projection,
+                                                        boolean newestSequence) {
+        return new LocalClusterMetaSupplier<>(
+                instance -> (EventJournalReader<EventJournalCacheEvent>) instance.getCacheManager().getCache(cacheName),
+                predicate, projection, newestSequence);
+    }
+
+    public static <T> ProcessorMetaSupplier streamCache(String cacheName,
+                                                        ClientConfig clientConfig,
+                                                        DistributedPredicate<EventJournalCacheEvent> predicate,
+                                                        DistributedFunction<EventJournalCacheEvent, T> projection,
+                                                        boolean newestSequence) {
+        return new RemoteClusterMetaSupplier<>(clientConfig,
+                instance -> (EventJournalReader<EventJournalCacheEvent>) instance.getCacheManager().getCache(cacheName),
+                predicate, projection, newestSequence);
+    }
+
+    @Override
+    protected void init(@Nonnull Context context) throws Exception {
+        super.init(context);
+        jobFuture = context.jobFuture();
+    }
+
+    private void subscribe(List<Integer> partitions) {
+        List<ICompletableFuture<EventJournalInitialSubscriberState>> futures = new ArrayList<>();
+        try {
+            for (Integer partitionId : partitions) {
+                futures.add(eventJournalReader.subscribeToEventJournal(partitionId));
+            }
+            for (int i = 0; i < futures.size(); i++) {
+                ICompletableFuture<EventJournalInitialSubscriberState> future = futures.get(i);
+                offsetMap.put(partitions.get(i), getSequence(future.get()));
+            }
+        } catch (Exception e) {
```

Comment:
should replace this with `uncheckedRun()` and limit it to where an exception is thrown rather than whole method